pytorch_lightning/loggers/comet.py

```diff
@@ -278,6 +279,9 @@ def version(self) -> str:
         if self._experiment_key is not None:
             return self._experiment_key
 
+        if "COMET_EXPERIMENT_KEY" in os.environ:
+            return os.environ["COMET_EXPERIMENT_KEY"]
+
```

Comment:
sorry for the question, I am not very familiar with this logger. 
if the env variable is supposed to determine the version, can it not be assigned to self._experiment_key, and then we use the above if statement to return the right version? What's the difference between self._experiment_key and the env variable?

Reply:
logic is based on the value of `self._experiment_key`  for example here https://github.com/PyTorchLightning/pytorch-lightning/blob/357de8e2bc610fcf32f96e5fed97b832493cbde2/pytorch_lightning/loggers/comet.py#L193
tldr if it is set it means that the experiment is already created and you should log in it. If not then create new experiment. Here we want a new experiment but with predefined experiment key, which is set with env variable

Reply:
Indeed the name is a bit confusing, the env variable `COMET_EXPERIMENT_KEY` is used to manually set the experiment key for new experiment. The `self._experiment_key` attribute is used to "continue" a previous experiment, it's passed as the `previous_experiment` parameter of `CometExistingExperiment`.

If the name don't need to be standard across loggers, `self._experiment_key` could be renamed `self._previous_experiment_key` in order to reduce confusion.

Reply:
Would be strange renaming internal variable and not renaming the api here. 
https://github.com/PyTorchLightning/pytorch-lightning/blob/357de8e2bc610fcf32f96e5fed97b832493cbde2/pytorch_lightning/loggers/comet.py#L125
If we are renaming to avoid confusion init parameter name should also be renamed to `previous_experiment_key`

Reply:
I guess all I was asking is, can we do

```python
self._experiment_key = os.environ.get("COMET_EXPERIMENT_KEY", experiment_key)
```
in init? (or maybe also warn/error to user if both are set)
Because as you explain, there are two ways to continue the experiment, either define the env var or pass in the key to the logger.

Reply:
As I've said that would affect the logic of the logger and ExistingExperiment will be used in this case which is not expected behaviour. To fix that you need to rethink logic based on `self._experiment_key`
Anyway I don't believe this is in scope of this pr